stats.go

```diff
@@ -0,0 +1,77 @@
+package tiledb
+
+/*
+#cgo LDFLAGS: -ltiledb
+#include <tiledb/tiledb.h>
+#include <stdio.h>
+#include <stdlib.h>
+*/
+import "C"
+
+import (
+	"fmt"
+	"os"
+	"unsafe"
+)
+
+// StatsEnable enable internal statistics gathering
+func StatsEnable() error {
+	ret := C.tiledb_stats_enable()
+	if ret != C.TILEDB_OK {
+		return fmt.Errorf("Error enabling stats")
+	}
+	return nil
+}
+
+// StatsDisable disable internal statistics gathering.
+func StatsDisable() error {
+	ret := C.tiledb_stats_disable()
+	if ret != C.TILEDB_OK {
+		return fmt.Errorf("Error disabling stats")
+	}
+	return nil
+}
+
+// StatsReset reset all internal statistics counters to 0.
+func StatsReset() error {
+	ret := C.tiledb_stats_reset()
+	if ret != C.TILEDB_OK {
+		return fmt.Errorf("Error resetting stats")
+	}
+	return nil
+}
+
+// StatsDumpSTDOUT prints internal stats to stdout
+func StatsDumpSTDOUT() error {
+	ret := C.tiledb_stats_dump(C.stdout)
+	if ret != C.TILEDB_OK {
+		return fmt.Errorf("Error dumping stats to stdout")
+	}
+	return nil
+}
+
+// StatsDump prints internal stats to file path given
+func StatsDump(path string) error {
+
+	if _, err := os.Stat(path); err == nil {
+		return fmt.Errorf("Error path already %s exists", path)
+	}
+
+	// Convert to char *
+	cPath := C.CString(path)
+	defer C.free(unsafe.Pointer(cPath))
+
+	// Set mode as char*
+	cMode := C.CString("w")
+	defer C.free(unsafe.Pointer(cMode))
+
+	// Open file to get FILE*
+	cFile := C.fopen(cPath, cMode)
```

Comment:
I think there needs to be a defer close here right?

Reply:
also if the dump is unsuccessful / throws an error for whatever reason, we should probably try to clean up the opened file

Reply:
Good catch, I've added the defer C.fclose call. When you say clean up the opened file, do you mean we should truncate/delete the file (if it exists) or do you just mean to make sure it gets closed? The defer C.fclose will close the file once the scope ends so it will close at either return.

Reply:
On second thought just ensuring the file is closed is fine.  I was thinking about if a file was created, but not written to this could be confusing.  However there are instances where it would not make sense to remove the unsuccessfully written file (ex a system file), so ignore that suggestion